lib/node_modules/@stdlib/assert/is-same-date-object/benchmark/benchmark.length.js

```diff
@@ -0,0 +1,50 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var isBoolean = require( '@stdlib/assert/is-boolean' ).isPrimitive;
+var pkg = require( './../package.json' ).name;
+var isSameDateObject = require( './../lib' );
+
+
+// FUNCTIONS //
+
+bench( pkg + '', function benchmark( b ) {
+	var date1 = new Date();
+	var date2 = new Date();
+	var bool;
+	var i;
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		date2.setMilliseconds( i ); // Change the date slightly each iteration
```

Comment:
I appreciate the intent to ensure we compare different dates for each iteration in order to avoid certain compiler optimizations. I think we can do better by simply using a pre-allocated list of `values` containing a list of date objects and then in the loop, just use cheap modulo arithmetic to pull out a new `date2` for comparison. If you hunt enough, you should be able to find a similar approach elsewhere in the project.

Reply:
got that, i'll make the changes and pull the 'date2' from a values array.

that aside i'm curious about your usage of the word 'cheap' , is there a high cost operation here ? if yes then which one and how. Would highly appreciate your knowledge

Reply:
A method call which mutates an object is likely to be slower than an operator (e.g., `%`) which is implemented in hardware. Here, we want to avoid doing operations which have perf costs similar to what we are attempting to test. Otherwise, perf results have conflating factors and it becomes harder to separate signal from noise when comparing across implementations and/or packages. Ideally, any boilerplate code around the operation of focus should be at least an order of magnitude faster, preferably more, such that any additional costs are negligible.